src/rgw/rgw_lua_utils.h

```diff
@@ -193,5 +193,101 @@ void set_package_path(lua_State* L, const std::string& install_dir);
 // and the "debug" library
 void open_standard_libs(lua_State* L);
 
+typedef int MetaTableClosure(lua_State* L);
+
+template<typename MapType>
+int StringMapWriteableNewIndex(lua_State* L) {
+  const auto map = reinterpret_cast<MapType*>(lua_touserdata(L, lua_upvalueindex(1)));
+
+  const char* index = luaL_checkstring(L, 2);
+  const char* value = luaL_checkstring(L, 3);
+  map->insert_or_assign(index, value);
+  return NO_RETURNVAL;
+}
+
+//Similar to StringMapWriteableNewIndex, however, a mutex is also passed here
```

Comment:
please move the lock inside the map class
lua does not need locking, so it should not be visible here